scripts/version_buckets.py

```diff
@@ -0,0 +1,243 @@
+                    "bucket": bucket_obj,
+                    "versioning": versioning.status,
+                    "lifecycle": lifecycle,
+                }
+    return result
+
+
+@click.group()
+@click.option("--verbose", is_flag=True, help="Comprehensive logging set to stdout.")
+def main(verbose=False):
+    """
+    A tool to manage the versioning and life cycle of S3 buckets.
+    """
+    if verbose:
+        global VERBOSE
+        VERBOSE = True
+        verbose_handler = logging.StreamHandler(sys.stdout)
+        verbose_handler.setLevel(logging.INFO)
+        verbose_handler.setFormatter(log_formatter)
+        logger.addHandler(verbose_handler)
+        click.echo("Logging to {}\n".format(LOGFILE))
+
+
+@main.command()
+@click.option("-n", "--number", required=False, type=int)
+def list(number=0):
+    """
+    List n (or all, if not given) bucket details.
+    """
+    buckets = get_buckets(number)
+    data = [("Name", "Versioned", "LifeCycle")]
+    col_width = [0, 0, 0]
+    rows = []
+    for name, bucket in buckets.items():
+        v = bucket["versioning"]
+        l = bucket["lifecycle"]
+        v = v if v else "Disabled"
+        if l:
+            l = json.dumps(l, indent=1)
+        else:
+            l = "None"
+        data.append((name, v, l))
+    for row in data:
+        for i, info in enumerate(row):
+            col_width[i] = min(max(len(info) + 2, col_width[i]), 48)
+    dashes = tuple(("-" * (width - 1) for width in col_width))
+    data.insert(1, dashes)
+    click.echo(f"The status of the buckets:")
+    for row in data:
+        output = ""
+        for i in range(3):
+            output += row[i].ljust(col_width[i])
+        if not VERBOSE:
+            click.echo(output)
+        logger.info(output)
+
+
+@main.command()
+@click.option("-n", "--number", required=False, type=int)
+@click.option("-x", "--execute", is_flag=True)
+def update(number=0, execute=False):
+    """
+    Update n (or all, if not given) bucket details. Use --execute to make the
+    changes happen, or else a preview will take place.
+    """
+    buckets = get_buckets(number)
+    for k, v in buckets.items():
+        name = k
+        bucket = v["bucket"]
+        versioning = v["versioning"] == "Enabled"
+        lifecycle = v["lifecycle"]
+        update_bucket(name, bucket, versioning, lifecycle, execute)
+
+
+def update_bucket(name, bucket, versioning, lifecycle, execute):
+    """
+    Update the bucket to the expected versioning and life-cycle settings.
+    """
+    msg = f"Working on {name}."
+    logger.info(msg)
+    if not VERBOSE:
+        click.echo(f"Working on {name}.")
+    # Add versioning if not already set.
+    if not versioning:
+        msg = f"Enabling versioning for {name}."
+        logger.info(msg)
+        if not VERBOSE:
+            click.echo(msg)
+        if execute:
+            # TODO: Uncomment to actually make the changes.
+            # v = bucket.Versioning()
+            # v.enable()
```

Comment:
This is the commented code I was referring to in other comment related to `--execute` flag.
If we default execute to `False` we may feel comfortable enough to have this uncommented (as it wouldn't run by default, unless of course the `--execute` flag is ignored).

Reply:
Yeah.... I won't uncomment this until ready. I just wanted to make sure anyone else running the command couldn't actually make any changes. Commenting out this (and the other) block of code ensures nothing accidental happens.